test/integration/executor/create_role_test.cpp

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "integration/executor/executor_fixture.hpp"
+
+#include <gtest/gtest.h>
+#include "common/result.hpp"
+#include "framework/common_constants.hpp"
+#include "integration/executor/executor_fixture_param_provider.hpp"
+#include "module/shared_model/mock_objects_factories/mock_command_factory.hpp"
+#include "module/shared_model/mock_objects_factories/mock_query_factory.hpp"
+
+using namespace common_constants;
+using namespace executor_testing;
+using namespace framework::expected;
+using namespace shared_model::interface::types;
+
+using shared_model::interface::permissions::Grantable;
+using shared_model::interface::permissions::Role;
+
+static const RoleIdType kAnotherRole("another_role");
+
+class CreateRoleTest : public BasicExecutorTest<ExecutorTestBase> {
+ public:
+  auto getRolePerms(const RoleIdType &role) {
+    return getItf().executeQueryAndConvertResult(
+        *getItf().getMockQueryFactory()->constructGetRolePermissions(role));
+  }
+
+  void checkRole(
+      const RoleIdType &role,
+      const shared_model::interface::RolePermissionSet &ref_permissions) {
+    getRolePerms(role).specific_response.match(
+        [&](const auto &test_permissions) {
+          EXPECT_EQ(test_permissions.value.rolePermissions(), ref_permissions)
+              << "Wrong set of permissions for role " << role;
+        },
+        [](const auto &e) { ADD_FAILURE() << e.error->toString(); });
+  }
+
+  void checkNoSuchRole(const RoleIdType &role) {
+    assertResultError(getRolePerms(role).specific_response);
+  }
+};
+
+/**
+ * @given a user with all kCreateRole permission
+ * @when executes CreateRole command
+ * @then the command succeeds and the role is created
+ */
+TEST_P(CreateRoleTest, Valid) {
+  getItf().createUserWithPerms(
+      kUser, kDomain, kUserKeypair.publicKey(), {Role::kCreateRole});
+  assertResultValue(getItf().executeCommandAsAccount(
+      *getItf().getMockCommandFactory()->constructCreateRole(
+          kAnotherRole, {Role::kCreateRole}),
+      kUserId,
+      true));
+  checkRole(kAnotherRole, {Role::kCreateRole});
+}
+
+/**
+ * @given a user with all kCreateRole permission
+ * @when executes CreateRole command with empty permission set
+ * @then the command succeeds and the role is created
```

Comment:
This case is stateless invalid. Looks like we need to connect stateless validation and execution in this framework, otherwise it is possible to write invalid cases.

Reply:
Bump?

Reply:
I don't think so. this is command and query executor backend test. it has its interface and it needs to be tested. if this test is out of scope of the command executor backend interface, I remove it.

Reply:
Alright, probably it makes sense in the context of configurable stateless validator.